src/app/common/components/Prompt.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { Modal, Button } from '@patternfly/react-core';
+import { useHistory } from 'react-router';
+import { noop } from '@app/utils/utils';
+import ConfirmModal from '@app/common/components/ConfirmModal';
+
+interface IPrompt {
+  when: boolean;
+  title: string;
+  message: string | JSX.Element;
+  okText?: string;
+  cancelText?: string;
+}
+
+const Prompt: React.FunctionComponent<IPrompt> = ({
+  when,
+  title,
+  message,
+  okText = 'OK',
+  cancelText = 'Cancel',
+}) => {
+  const history = useHistory();
+  const [showPrompt, setShowPrompt] = React.useState(when);
+  const [currentPath, setCurrentPath] = React.useState('');
+
+  React.useEffect(() => {
+    if (when) {
+      history.block((prompt) => {
+        setCurrentPath(prompt.pathname);
+        setShowPrompt(true);
+        return 'true';
+      });
+    }
+  }, [history, when]);
+
+  const handleOk = React.useCallback(() => {
+    history.block(noop);
```

Comment:
Maybe if we store the `unblock` function from above in a `useRef`, we can just call it here instead of having to pass a noop to clear the block which doesn't look like documented behavior as far as I can tell from the docs.

Reply:
Ok I pushed some updates, what's strange is that the block function doesn't actually return anything usable as mentioned in the docs. For now I stored a similar noop style function within the component scope that serves the same purpose and is used in the same way.